pkg/apis/rollouts/v1alpha1/types.go

```diff
@@ -196,8 +198,44 @@ type CanaryStep struct {
 	Pause *RolloutPause `json:"pause,omitempty"`
 	// Experiment defines the experiment object that should be created
 	Experiment *RolloutExperimentStep `json:"experiment,omitempty"`
+	// AnalysisRun defines the AnalysisRun that will run for a step
+	AnalysisRun *RolloutAnalysisRun `json:"analysisRun,omitempty"`
```

Comment:
So we have the field name `parallelAnalysis` above and `analysisRun` here, which is inconsistent, despite reusing the same spec. I was thinking it should be just `analysis` in both cases. Keep in mind we'll also need a field for this in the Experiment spec.

```yaml
  strategy:
    canary: 
      analysis:
        templateName: success-rate
```

```yaml
  strategy:
    canary: 
      steps:
      - setWeight: 20
      - pause: {duration: 300}
      - analysis:
          templateName: success-rate
```